db/csv/farmers_market_generator.rb

```diff
@@ -0,0 +1,92 @@
+require 'csv'
+
+class FarmersMarketGenerator
+  def initialize
+    @csv = File.open('db/csv/markets.csv')
+  end
+
+  def import
+    create_products
+    create_markets
+  end
+
+  private
+
+  def create_markets
+    CSV.foreach(@csv, headers: true, header_converters: :symbol) do |row|
+      csv_market = row.to_h
+      market = Market.create(market_params(csv_market))
+      create_market_products(csv_market, market)
+    end
+  end
+
+  def create_market_products(csv_market, market)
+    product_ids(csv_market).each do |id|
+      MarketProduct.create(market_id: market.id, product_id: id)
+    end
+  end
+
+  def product_ids(csv_market)
+    Product.where(name: market_products(csv_market)).ids
+  end
+
+  def market_products(csv_market)
+    csv_market.select { |k, v| v == 'Y' && product_list.include?(k) }.keys
+  end
+
+  def create_products
```

Comment:
so this would create products with a symbol as the name? do they need to be symbols or does it matter?

Reply:
Since we specify that column type is `string`, these symbols get converted to strings upon creation. Having the `product_list` as symbols makes comparing/finding the corresponding products for each market much easier/more efficient.